src/Service.Host/client/src/components/Pallets.js

```diff
@@ -0,0 +1,596 @@
+        isLoading: createPalletLoading,
+        errorMessage: createError,
+        postResult: createResult,
+        clearPostResult: clearCreateResult
+    } = usePost(itemTypes.pallets.url);
+
+    useEffect(() => {
+        if (updateResult) {
+            setPallet(updateResult);
+            setSnackbarVisible(true);
+            clearUpdateResult();
+        } else if (createResult) {
+            setPallet(createResult);
+            setSnackbarVisible(true);
+            clearCreateResult();
+        }
+    }, [updateResult, createResult, clearCreateResult, clearUpdateResult]);
+
+    const [pendingSelected, setPendingSelected] = useState(null);
+
+    const handlePalletSearchResultSelect = selected => {
+        if (selected && selected.palletNumber) {
+            getInitialDepartment(selected.auditedByDepartmentCode);
+            setPendingSelected(selected);
+        }
+    };
+
+    useEffect(() => {
+        if (pendingSelected && departmentGetResult) {
+            const auditedByEmployee =
+                historicalEmployeesResult?.items.find(
+                    emp => emp.id === pendingSelected?.auditedBy
+                ) || {};
+
+            setPallet({
+                ...pendingSelected,
+                auditedByEmployee: auditedByEmployee.fullName,
+                storageLocationDescription: pendingSelected.storageLocation?.description,
+                defaultStockPoolDescription: pendingSelected.defaultStockPool?.stockPoolDescription,
+                departmentCode: departmentGetResult?.departmentCode,
+                auditedByDepartmentDescription: departmentGetResult?.description
+            });
+            setPendingSelected(null); // This prevents a loop!
+        }
+    }, [pendingSelected, departmentGetResult, historicalEmployeesResult]);
+
+    const handleDepartmentSearchResultSelect = selected => {
+        setPallet({
+            ...pallet,
+            auditedByDepartmentCode: selected.departmentCode,
+            auditedByDepartmentDescription: selected.description
+        });
+    };
+    const handleStockPoolSearchResultSelect = selected => {
+        setPallet({
+            ...pallet,
+            defaultStockPool: selected,
+            defaultStockPoolId: selected.stockPoolCode,
+            defaultStockPoolDescription: selected.stockPoolDescription
+        });
+    };
+
+    const handleStorageLocationSearchResultSelect = selected => {
+        setPallet({
+            ...pallet,
+            storageLocation: selected,
+            storageLocationId: selected.locationId,
+            storageLocationDescription: selected.locationCode
+        });
+    };
+
+    const handleEmployeeFieldChange = (propertyName, newValue) => {
+        const employeeInfo = currentEmployeesResult?.items.find(emp => emp.fullName === newValue);
+
+        setPallet(c => ({
+            ...c,
+            auditedBy: employeeInfo?.id,
+            auditedByEmployee: employeeInfo?.fullName
```

Comment:
you shouldn't need to do this (manually specify the name field to display) - You should configure your Dropdown's items list to be an array of objects with 'id' and 'displayText' properties. Then your form state only needs to track the id of the selected option and the dropdown will be smart enough to render the correct name automatically without you having to look it up from the employees list again